src/WebJobs.Extensions.Http/Utility.cs

```diff
@@ -26,7 +26,7 @@ internal static void ApplyBindingData(object value, Dictionary<string, object> b
                     // parse the object adding top level properties
                     JObject parsed = JObject.Parse(json);
                     var additionalBindingData = parsed.Children<JProperty>()
-                        .Where(p => p.Value != null && (p.Value.Type != JTokenType.Array))
+                        .Where(p => p.Value != null)
```

Comment:
The code comment above indicates that we're intentionally only pulling binding data from top level properties. We don't want to add JArray properties like Length, etc. to binding data